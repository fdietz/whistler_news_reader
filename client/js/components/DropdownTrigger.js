import React, { PropTypes } from 'react';
import classNames from 'classnames';

const DropdownTrigger = ({ children, className, style, active, onClick, ...rest }) => {
  const cls = classNames('dropdown__trigger', className, {
    active: active,
  });

  return (
    <a href="#" className={cls} onClick={onClick} style={style} {...rest}>
      {children}
    </a>
  );
};

DropdownTrigger.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object,
  active: PropTypes.bool,
  onClick: PropTypes.func,
};

export default DropdownTrigger;
