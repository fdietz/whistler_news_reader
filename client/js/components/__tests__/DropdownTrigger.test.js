import renderer from 'react-test-renderer';
import React from 'react';
import DropdownTrigger from '../DropdownTrigger';

describe('DropdownTrigger', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DropdownTrigger />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
