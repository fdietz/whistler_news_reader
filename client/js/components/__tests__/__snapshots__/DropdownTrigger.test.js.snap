exports[`DropdownTrigger renders correctly 1`] = `
<a
  className="dropdown__trigger"
  onClick={undefined}
  style={undefined} />
`;
